tests/protocols/test_http3.py

```
@@ -0,0 +1,785 @@
+    harness, protocol, _ = await connected(echo_app())
+    harness.send_request(b"GET", b"/")
+    await harness.pump()
+    assert protocol in protocol.server_state.connections
+    protocol.shutdown()
+    # No live connections remain, so the endpoint deregisters and closes its socket.
+    assert protocol not in protocol.server_state.connections
+    assert protocol.transport.is_closing()
+
+
+async def test_http3_shutdown_refuses_new_connections() -> None:
+    protocol, transport, _ = make_protocol(echo_app())
```

Ruff (0.16.1)

[warning] Unpacked variable `transport` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)